src/main/java/mat/client/cqlworkspace/AbstractCQLWorkspacePresenter.java

```diff
@@ -494,7 +494,7 @@ public void onSuccess(VsacApiResult result) {
 
 	protected void addCQLLibraryEditorViewHandlers() {
 		cqlWorkspaceView.getCQLLibraryEditorView().getExportErrorFile().addClickHandler(event -> exportErrorFile());
-		cqlWorkspaceView.getCQLLibraryEditorView().getInsertButton().addClickHandler(event -> buildInsertPopUp());
+		cqlWorkspaceView.getCQLLibraryEditorView().getInsertButton().addClickHandler(event -> buildInsertPopUp(MatContext.get().getCurrentMeasureModel()));

```

Comment:
how is this compatible with standalone cql workspace implementation? I'm not sure if a standalone library can rely on a current measure model type.
should it use MatContext.get().getCurrentCQLLibraryModelType()) instead?

Reply:
addCQLLibraryEditorViewHandlers() is being used by both stand-alone and measure grid. It registers the event handlers when starting the application. It's an arbitrary situation.

Reply:
Right. But standalone CQL should rely on MatContext.get().getCurrentCQLLibraryModelType() instead of MatContext.get().getCurrentMeasureModel()

How about this.. 
Does it make sense to define 
abstract String AbstractCqlWorkspacePresenter::getCurrentModelType 

CQLMeasureWorkSpacePresenter::getCurrentModelType = () -> MatContext.get().getCurrentMeasureModel()
CQLStandaloneWorkSpacePresenter::getCurrentModelType = () -> MatContext.get().getCurrentCQLLibraryModelType()

Then, you can use getCurrentModelType in all 3 presenters

Reply:
Implemented